Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -317,43 +318,33 @@ class Surface : public virtual GeometryObject,
                                                 const Vector3D& position,
                                                 const Vector3D& momentum) const;
 
-  /// Initialize the jacobian from local to global
-  /// the surface knows best, hence the calculation is done here.
-  /// The jacobian is assumed to be initialised, so only the
-  /// relevant entries are filled
+  /// Calculate the jacobian from local to global which the surface knows best,
+  /// hence the calculation is done here
   ///
   /// @todo this mixes track parameterisation and geometry
   /// should move to :
   /// "Acts/EventData/detail/coordinate_transformations.hpp"
   ///
   /// @param gctx The current geometry context object, e.g. alignment
-  /// @param jacobian is the jacobian to be initialized
-  /// @param position is the global position of the parameters
-  /// @param direction is the direction at of the parameters
-  /// @param pars is the parameter vector
-  virtual void initJacobianToGlobal(const GeometryContext& gctx,
-                                    BoundToFreeMatrix& jacobian,
-                                    const Vector3D& position,
-                                    const Vector3D& direction,
-                                    const BoundVector& pars) const;
-
-  /// Initialize the jacobian from global to local
-  /// the surface knows best, hence the calculation is done here.
-  /// The jacobian is assumed to be initialised, so only the
-  /// relevant entries are filled
+  /// @param boundParams is the bound parameters vector
+  ///
+  /// @return Jacobian from local to global
+  virtual BoundToFreeMatrix jacobianLocalToGlobal(
+      const GeometryContext& gctx, const BoundVector& boundParams) const;
+
+  /// Calculate the jacobian from global to local which the surface knows best,
+  /// hence the calculation is done here
   ///
   /// @todo this mixes track parameterisation and geometry
   /// should move to :
   /// "Acts/EventData/detail/coordinate_transformations.hpp"
   ///
-  /// @param jacobian is the jacobian to be initialized
-  /// @param position is the global position of the parameters
-  /// @param direction is the direction at of the parameters
   /// @param gctx The current geometry context object, e.g. alignment
-  virtual void initJacobianToLocal(const GeometryContext& gctx,
-                                   FreeToBoundMatrix& jacobian,
-                                   const Vector3D& position,
-                                   const Vector3D& direction) const;
+  /// @param parameters is the free parameters
+  ///
+  /// @return Jacobian from global to local
+  virtual FreeToBoundMatrix jacobianGlobalToLocal(
+      const GeometryContext& gctx, const FreeVector& parameters) const;
```

Comment:
Just for my understanding: what happens if the input parameters are not on the surface? Should this return a `Result` or do we assume that this is handled on a higher level?

Reply:
We just assumes the input parameter here is on surface. I added some description.

Reply:
Ok, thanks.